### PR TITLE
feat(remote): add ssh remote peapod

### PR DESCRIPTION
**Pull request by hanxiao**
Reason for this PR:
- we can have multiple remote access methods
- having an SSH-based remote peapod requires no extra dependency, long-term: ssh peapod is a poorman solution for novice
- help us to design better RemotePea/Pod lifecycle

Not trying to overengineer anything at the moment.

### CODE REVIEW

File: jina/peapods/ssh.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+from subprocess import Popen, PIPE
+
+from .. import __ready_msg__, __stop_msg__
+from ..helper import get_non_defaults_args, kwargs2list
+from ..logging import JinaLogger
+from ..peapods.pea import BasePea
+
+
+class RemotePea(BasePea):
```
Comment (JoanFM) [inline jina/peapods/ssh.py:9]: Isnt this name already used in another file? Can we have RemoteSSHPea?

File: jina/peapods/ssh.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    @property
+    def remote_command(self) -> str:
+        from ..parser import set_pod_parser
+        non_defaults = get_non_defaults_args(self.args, set_pod_parser(), taboo={'host'})
+        _args = kwargs2list(non_defaults)
+        return f'jina pod {" ".join(_args)}'
+
+
+class RemoteMutablePod(RemotePod):
```
Comment (JoanFM) [inline jina/peapods/ssh.py:60]: Same here. It can lead to confusion?

**Comment by JoanFM** [comment]
I think that this PR at this moment when we are stabilizing our distribution vision (and when potentially we may rearrange the Pods and Peas) adds not so much value and can add maintenance cost and may limit some redesign.

**Comment by hanxiao** [comment]
> and when potentially we may rearrange the Pods and Peas

as the refactoring work has not started yet, it does not block any existing work. Having SSH peapod helps us to design a better refactoring on RemotePea/Pod.

File: jina/peapods/ssh.py (hunk 2 above)
Comment (hanxiao) [reply jina/peapods/ssh.py:60]: they are in different module

Comment (JoanFM) [reply jina/peapods/ssh.py:60]: Yes but leads to confusion

**Comment by hanxiao** [comment]
One thing I'd like to highlight about RemotePod is that we don't need to manually implement close/kill method on the remote. Simply send `terminate` signal via control port, this is exactly why it is inherited from `BasePea`.

**Pull request merged**